lib/mix/lib/mix/utils.ex

```diff
@@ -15,7 +15,31 @@ defmodule Mix.Utils do
   stored there.
   """
   def mix_home do
-    System.get_env("MIX_HOME") || Path.expand("~/.mix")
+    case {System.get_env("MIX_HOME"), System.get_env("XDG_DATA_HOME")} do
+      {directory, _} when is_binary(directory) -> directory
+      {nil, directory} when is_binary(directory) -> :filename.basedir(:user_data, "mix")
+      {nil, nil} -> Path.expand("~/.mix")
+    end
+  end
+
+  @doc """
+  Gets possible locations of global Mix configuration.
+
+  Possible locations:
+
+   * `~/.mix`
+   * `MIX_HOME`
+   * `XDG_DATA_HOME/mix`
+
+  """
+  def mix_config_homes do
```

Comment:
I think this can be the same  as `mix_home`, i.e. we will just lookup at one of the directories, depending the variable is set. In fact, this change is even backwards incompatible as it can  load multiple configs thaat wouldn't be loaded today.